desktop-widgets/divelistview.cpp

```diff
@@ -424,7 +411,7 @@ void DiveListView::reload(DiveTripModel::Layout layout, bool forceSort)
 
 	QSortFilterProxyModel *m = qobject_cast<QSortFilterProxyModel *>(model());
 	QAbstractItemModel *oldModel = m->sourceModel();
-	tripModel = new DiveTripModel(this);
+	DiveTripModel *tripModel = new DiveTripModel(this);
```

Comment:
memory leak?
`DiveListView::reload()` is called more than once.

Reply:
No, the old model is deleted a few lines later (that's the reason for fetching `oldModel`).

But you are right: this code is really shoddy. I have a commit to replace this with a "singleton". Though since the object is only used in this class, we might even make it a sub-object.